tsl/test/src/compression_unit_test.c

```diff
@@ -1069,9 +1088,250 @@ test_simple8b_rle()
 	}
 }
 
+static void
+compare_datum_ptr(Datum datum1, Datum datum2, const char *msg)
+{
+	char *ptr1 = DatumGetPointer(datum1);
+	char *ptr2 = DatumGetPointer(datum2);
+	size_t size1 = VARSIZE(datum1);
+	size_t size2 = VARSIZE(datum2);
+	TestAssertTrue(ptr1 != NULL);
```

Comment:
I think this should be before the VARSIZE check otherwise it risks being optimized away as UB.